compiler/src/dmd/declaration.d

```diff
@@ -1896,11 +1902,12 @@ extern (C++) class BitFieldDeclaration : VarDeclaration
 
         if (!fieldState.inFlight)
         {
+            //printf("not in flight\n");
             startNewField();
         }
         else if (style == TargetC.BitFieldStyle.Gcc_Clang)
         {
-            if (fieldState.bitOffset + fieldWidth > memsize * 8)
+            if (fieldState.bitOffset + fieldWidth > memsize * 8 && t.ty != Tbool)
```

Comment:
This should be advance to the next boundary only if the field at bitOffset would span more units of _alignment_ than its type.

Reply:
Something like the following.
```suggestion
            // At what offset relative to the underlying type is the bit-field being added.
            const bitOffset = fieldState.bitOffset % (memalignsize * 8);
            // If the bit-field spans more units of alignment than its type,
            // start a new field at the next alignment boundary.
            if (bitOffset + fieldWidth > memsize * 8)
```

Reply:
merged